azkaban-web-server/src/main/java/cloudflow/error/CloudFlowException.java

```diff
@@ -0,0 +1,23 @@
+package cloudflow.error;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class CloudFlowException extends Exception {
+
+    // A list to avoid multiple api calls when there are various validation errors. Especially
+    // useful with endpoints that take several parameters.
+    private List<CloudFlowError> errors;
```

Comment:
I haven't seen the use of lists of errors in current Azkaban code but I think it makes sense for CloudFlow. Also, I don't think there is a formal requirement for this. The intention of having a list of errors is basically to communicate multiple validation errors in a single call, reducing the number of api calls and leading to a better user experience and less iterations. This is especially useful when creating new executions. I agree that we should define its usage properly since it's optional and subjective. I'm open to discuss this further.